lighthouse-core/scripts/i18n/collect-strings.js

```diff
@@ -375,6 +389,60 @@ function createPsuedoLocaleStrings(messages) {
   return psuedoLocalizedStrings;
 }
 
+/**
+ * @param {import('typescript').NamedDeclaration} node
+ * @return {string}
+ */
+function getIdentifier(node) {
+  if (!node.name || !tsc.isIdentifier(node.name)) throw new Error('no Identifier found');
+
+  return node.name.text;
+}
+
+/**
+ * @param {string} sourceStr String of the form 'const UIStrings = {...}'.
+ * @param {Record<string, string>} liveUIStrings The actual imported UIStrings object.
+ * @return {Record<string, ParsedUIString>}
+ */
+function parseUIStrings(sourceStr, liveUIStrings) {
+  const tsAst = tsc.createSourceFile('uistrings', sourceStr, tsc.ScriptTarget.ES2019, true, tsc.ScriptKind.JS);
+
+  const extractionError = new Error('UIStrings was not extracted correctly by the collect-strings regex.');
+  const uiStringsStatement = tsAst.statements[0];
+  if (tsAst.statements.length !== 1) throw extractionError;
+  if (!tsc.isVariableStatement(uiStringsStatement)) throw extractionError;
+
+  const uiStringsDeclaration = uiStringsStatement.declarationList.declarations[0];
+  if (!tsc.isVariableDeclaration(uiStringsDeclaration)) throw extractionError;
+  if (getIdentifier(uiStringsDeclaration) !== 'UIStrings') throw extractionError;
+
+  const uiStringsObject = uiStringsDeclaration.initializer;
+  if (!uiStringsObject || !tsc.isObjectLiteralExpression(uiStringsObject)) throw extractionError;
```

Comment:
we could alternatively drop a lot of the 9 lines above, set
```js
let uiStringsObject;
try {
  // @ts-ignore
  uiStringsObject = tsAst.statements[0].declarationList.declarations[0].initializer;
  if (tsc.isObjectLiteralExpression(uiStringsObject)) throw new Error('bad');
} catch (e) {
  throw new Error('UIStrings was not extracted correctly by the collect-strings regex.');
}
```
The type narrowing as we walk down is of questionable value since
a) we created the parsed string ourselves
b) we throw on any and all assumption failures anyway
c) meanwhile it looks more important than it is

Reply:
eh I kinda like the way it explains the structure of everything, so I'm for keeping as-is now that we have it 👍